src/main/java/frc/robot/utils/ThreadedGyro.java

```diff
@@ -0,0 +1,56 @@
+package frc.robot.utils;
+
+import com.kauailabs.navx.frc.AHRS;
+import frc.robot.constants.Constants;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class ThreadedGyro {
+    private final AHRS gyro;
+    private final AtomicBoolean shouldReset = new AtomicBoolean(false);
+    private final AtomicBoolean shouldOffset = new AtomicBoolean(false);
+    private final AtomicLong lastGyro;
+    private final AtomicLong gyroOffset = new AtomicLong();
+    private final ScheduledExecutorService executor;
+
+    public ThreadedGyro(AHRS gyro) {
+        this.gyro = gyro;
+        this.lastGyro = new AtomicLong((Double.doubleToLongBits(0)));
+        this.executor = Executors.newScheduledThreadPool(1);
+    }
+
+    public void start(){
+        executor.scheduleAtFixedRate(() -> {
```

Comment:
We discussed getting the value synchronously prior to starting the thread to ensure a valid reading (reminder)

Reply:
done (in new PR)